tensorflow/examples/how_tos/reading_data/fully_connected_reader.py

```diff
@@ -143,47 +149,33 @@ def run_training():
                        tf.local_variables_initializer())
 
     # Create a session for running operations in the Graph.
-    sess = tf.Session()
-
-    # Initialize the variables (the trained variables and the
-    # epoch counter).
-    sess.run(init_op)
-
-    # Start input enqueue threads.
-    coord = tf.train.Coordinator()
-    threads = tf.train.start_queue_runners(sess=sess, coord=coord)
-
-    try:
-      step = 0
-      while not coord.should_stop():
-        start_time = time.time()
-
-        # Run one step of the model.  The return values are
-        # the activations from the `train_op` (which is
-        # discarded) and the `loss` op.  To inspect the values
-        # of your ops or variables, you may include them in
-        # the list passed to sess.run() and the value tensors
-        # will be returned in the tuple from the call.
-        _, loss_value = sess.run([train_op, loss])
-
-        duration = time.time() - start_time
-
-        # Print an overview fairly often.
-        if step % 100 == 0:
-          print('Step %d: loss = %.2f (%.3f sec)' % (step, loss_value,
+    with tf.Session().as_default() as sess:
```

Comment:
Optional: this `as_default()` is unnecessary.

Reply:
Actually, I'd upgrade this to "Mandatory:", since the only difference in adding `.as_default()` is to delay closing the `Session` until it is garbage collected. In this code, it might not make a difference, but if somebody copies and pastes it (as they are likely to do) they might find that the session stays alive for longer than they'd like.

Reply:
Oh, I wasn't even aware of that behavior regarding garbage collection. I've picked it up after carelessly calling `eval()` on a tensor once and the error suggested:

    No default session is registered. Use `with sess.as_default()` or pass an explicit session to 
    `eval(session=sess)`

I'll change that ... here and in my own code. Very good catch, thanks.